### DISCUSSION TITLE
Copilot + Calva integration

### DISCUSSION THREAD

**Discussion by adamhaber** [community/community#7413] [Copilot Conversations]
Hi,

I'm using Copilot to write Clojure within VSCode and I am amazed by how well it predicts what I actually want to write. One thing that I wasn't able to resolve is the integration with Calva (VSCode Clojure extension). When the Calva extension is enabled, Copilot's TAB completion doesn't work for me. Is there a way to fix this?

**Comment by dmrd** [comment]
+1 - the suggestions for clojure(script) are generally pretty good, but it is not possible to actually use them because of some interaction with Calva, the standard clojure layer for vscode.

Also created issue in calva repo: https://github.com/BetterThanTomorrow/calva/issues/1461

**Comment by bpringe** [comment]
Calva maps <kbd>tab</kbd> to its command for formatting the current form. As a workaround, you can map that command to some other key in the keyboard shortcuts or simply remove the mapping (formatting with a keyboard shortcut is pretty handy though 😄 ).
